pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -553,6 +574,15 @@ func (ctrl *Controller) syncContainerRuntimeConfig(key string) error {
 			if err != nil {
 				return ctrl.syncStatusOnly(cfg, err, "could not create MachineConfig from new Ignition config: %v", err)
 			}
+			_, ok := cfg.GetAnnotations()[ctrlcommon.MCNameSuffixAnnotationKey]
+			arr := strings.Split(managedKey, "-")
+			// If the MC name suffix annotation does not exist and the maaged key value returned has a suffix, then add the MC name
+			// suffix annotation and suffix value to the ctrcfg object
+			if len(arr) > 4 && !ok {
```

Comment:
why is len 4? probably obvi but I dont understand

Reply:
The first MC name would be `99-generated-[pool]-containerruntime` and the following ones will have a suffix e.g `99-generated-[pool]-containerruntime-1` so we want to only add the suffix annotation if a suffix exists that is why we split at `-` and check if the length is greater than 4.